crates/api_crud/src/post/create.rs

```diff
@@ -192,3 +191,35 @@ impl PerformCrud for CreatePost {
     .await
   }
 }
+
+/// Check the intersection of user languages and community languages. If it contains only one item,
+/// use that as default language for the post. Otherwise, use "undetermined".
+async fn default_post_language(
+  local_user_id: LocalUserId,
+  community_id: CommunityId,
+  pool: &DbPool,
+) -> Result<LanguageId, LemmyError> {
+  let (user_langs, community_langs, undetermined_lang) = blocking(pool, move |conn| {
+    Ok::<(Vec<LanguageId>, Vec<LanguageId>, LanguageId), LemmyError>((
+      LocalUserLanguage::read(conn, local_user_id)?,
+      CommunityLanguage::read(conn, community_id)?,
+      Language::undetermined(),
```

Comment:
Still an issue, this intersect logic should be done in SQL.

Reply:
I didnt finish addressing all the review comments yet, thats why they arent marked as resolved. You dont need to comment the same things again.